Add new example integration with Notion

leimonio opened the pull request:
Hi 👋 Just wanted to check with you what would you thing of this new example integration with Notion API.
I'd still need to update `README.dm` with more details.

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.

## Documentation / Examples

- [ ] Make sure the linting passes

leerob commented on examples/cms-notion/README.md:
```diff
@@ -0,0 +1,172 @@
+# A statically generated blog example using Next.js and Prismic
```
Please update the README 👍

leerob commented on examples/cms-notion/components/avatar.js:
```diff
@@ -0,0 +1,12 @@
+export default function Avatar({ name, picture }) {
+  return (
+    <div className="flex items-center">
+      <img
```
Could you start with one of the CMS examples that uses `next/image` or update this?

leerob commented on examples/cms-notion/components/cover-image.js:
```diff
@@ -0,0 +1,25 @@
+
+export default function CoverImage({ title, url, slug }) {
+  const image = (
+    <img
```
Could you start with one of the CMS examples that uses `next/image` or update this?

leerob commented on examples/cms-notion/components/cover-image.js:
```diff
@@ -0,0 +1,25 @@
+  return (
+    <div className="sm:mx-0">
+      {slug ? (
+        <Link as={`/posts/${slug}`} href="/posts/[slug]">
```
Could you start with one of the CMS examples that has `as` removed or update this? It's no longer needed.

leerob commented on examples/cms-notion/components/date.js:
```diff
@@ -0,0 +1,7 @@
+// import { Date as PrismicDate } from 'prismic-reactjs'
+
+export default function Date({ dateString }) {
+  // const date = PrismicDate(dateString)
```
Should we move this PR to a draft state?

leerob commented on examples/cms-notion/components/hero-post.js:
```diff
@@ -0,0 +1,44 @@
+      <div className="md:grid md:grid-cols-2 md:col-gap-16 lg:col-gap-8 mb-20 md:mb-28">
+        <div>
+          <h3 className="mb-4 text-4xl lg:text-6xl leading-tight">
+            <Link as={`/posts/${slug}`} href="/posts/[slug]">
```
Another 👍

leerob commented on examples/cms-notion/lib/api.js:
```diff
@@ -0,0 +1,24 @@
+
+export const server = dev ? 'http://localhost:3000' : 'https://your_deployment.server.com';
+
+export async function getAllPostsForHome() {
```
These don't need to be behind API routes, as we can use the logic directly since it's on the server 👍

leimonio replied:
Do you mean moving the logic from here to the api server, in order to minimise the different locations of the response data?

leerob commented on examples/cms-notion/package.json:
```diff
@@ -0,0 +1,23 @@
+    "classnames": "2.2.6",
+    "date-fns": "2.10.0",
+    "next": "latest",
+    "react": "^16.13.0",
```
You'll need React 17 with latest Next.js

leerob commented on examples/cms-notion/package.json:
```diff
@@ -0,0 +1,23 @@
+  "devDependencies": {
+    "@fullhuman/postcss-purgecss": "^2.1.0",
+    "postcss-preset-env": "^6.7.0",
+    "tailwindcss": "^1.2.0"
```
Could you update Tailwind to use JIT?

leerob commented:
I don't believe the Notion API supports enough block types to match the other CMS examples, right? (e.g. no support for images yet).

leerob commented:
I don't believe the Notion API supports enough block types to match the other CMS examples, right? (e.g. no support for images yet) 🙏

leimonio commented:
hey @leerob 👋 wow that was blazing fast feedback! kudos 💯

Indeed Notion support of block types is pretty basic, I wouldn't consider it actually a CMS, but can act-as-a CMS. 
I can find great value of an integration of Notion with NextJS. Currently Notion API is in Beta, and this can be a good example to start with as more and more people are using Notion these days. I personally like the idea of adding a new blog post even from my mobile and that can easily become available in my blog.

leerob commented:
Indeed, I love Notion, but I can't really recommend it for a full-featured blog just yet. Very soon hopefully 😄

leimonio commented:
@leerob I've addressed the comments above, thank you very much for the feedback! 💯

Renamed the folder to `with-notion` so as not to confuse it with full featured cms options. Also added a disclaimer in the README, so as user of the integration are aware of it.

Please have a look and let me know if there is anything else you'd like me to address before adding this example.

leerob commented on examples/with-notion/README.md:
```diff
@@ -0,0 +1,137 @@
+
+After creating an account, create a new \*page** and then a database as **full page table\*\* from the left panel.
+
+### Step 2. Create database columns
```
Could you just link a Notion page they can clone?

leerob commented on examples/with-notion/README.md:
```diff
@@ -0,0 +1,137 @@
+
+## Demo
+
+### [https://next-blog-prismic.vercel.app/](https://next-blog-prismic.vercel.app/)
```
Oops

balazsorban44 commented:
Hi @leimonio, would you be able to bring this PR up to date with our [examples guidelines](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)? :pray:

balazsorban44 reviewed (changes requested):
Nice work! Hi @leimonio, would you be able to bring this PR up to date with our [examples guidelines](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md) and address any feedback? :pray:

If not, that is alright as well, let's close it for now then. :green_heart:

leerob commented:
Closing this as stale, but open to reviewing a new PR if you're interested in continuing this forward!
